examples/pytorch/ogb/directional_GSN/main.py

```diff
@@ -0,0 +1,433 @@
+            h = h_t
+
+        g.ndata['h'] = h
+
+        hg = dgl.mean_nodes(g, 'h')
+
+        return self.MLP_layer(hg)
+
+    def loss(self, scores, labels):
+        is_labeled = labels == labels
+        loss = torch.nn.BCEWithLogitsLoss()(scores[is_labeled], labels[is_labeled].float().to('cuda'))
+        return loss
+
+def train_epoch(model, optimizer, device, data_loader):
+    model.train()
+    epoch_loss = 0
+    epoch_train_AP = 0
+    list_scores = []
+    list_labels = []
+    for iter, (batch_graphs, batch_labels, batch_snorm_n, batch_snorm_e) in enumerate(data_loader):
+        batch_x = batch_graphs.ndata['feat'].to(device)  # num x feat
+        batch_e = batch_graphs.edata['feat'].to(device)
+        batch_snorm_e = batch_snorm_e.to(device)
+        batch_snorm_n = batch_snorm_n.to(device)
+        batch_labels = batch_labels.to(device)
+        batch_graphs = batch_graphs.to(device)
+        optimizer.zero_grad()
+
+        batch_scores = model(batch_graphs, batch_x, batch_e, batch_snorm_n)
+        
+        loss = model.loss(batch_scores, batch_labels)
+        loss.backward()
+        optimizer.step()
+        epoch_loss += loss.detach().item()
+        list_scores.append(batch_scores.detach())
+        list_labels.append(batch_labels.detach())
+
+    epoch_loss /= (iter + 1)
+
+    evaluator = Evaluator(name='ogbg-molpcba')
+    epoch_train_AP = evaluator.eval({'y_pred': torch.cat(list_scores),
+                                        'y_true': torch.cat(list_labels)})['ap']
+
+    return epoch_loss, epoch_train_AP, optimizer
+
+def evaluate_network(model, device, data_loader):
+    model.eval()
+    epoch_test_loss = 0
+    epoch_test_AP = 0
+    with torch.no_grad():
+        list_scores = []
+        list_labels = []
+        for iter, (batch_graphs, batch_labels, batch_snorm_n, batch_snorm_e) in enumerate(data_loader):
+            batch_x = batch_graphs.ndata['feat'].to(device)
+            batch_e = batch_graphs.edata['feat'].to(device)
+            batch_snorm_e = batch_snorm_e.to(device)
+            batch_snorm_n = batch_snorm_n.to(device)
+            batch_labels = batch_labels.to(device)
+            batch_graphs = batch_graphs.to(device)
+
+            batch_scores = model.forward(batch_graphs, batch_x, batch_e, batch_snorm_n)
+
+            loss = model.loss(batch_scores, batch_labels)
+            epoch_test_loss += loss.detach().item()
+            list_scores.append(batch_scores.detach())
+            list_labels.append(batch_labels.detach())
+
+        epoch_test_loss /= (iter + 1)
+
+        evaluator = Evaluator(name='ogbg-molpcba')
+        epoch_test_AP = evaluator.eval({'y_pred': torch.cat(list_scores),
+                                           'y_true': torch.cat(list_labels)})['ap']
+
+    return epoch_test_loss, epoch_test_AP
+
+def view_model_param(net_params):
+    model = DGNNet(net_params.hidden_dim, net_params.out_dim, net_params.dropout, net_params.L, net_params.avg_d)
```

Comment:
Either you can drop this whole function or move it to `train`. It does not make sense to construct a model object purely for printing.